src/dmd/globals.d

```diff
@@ -718,4 +722,50 @@ enum PINLINE : ubyte
 
 alias StorageClass = uinteger_t;
 
+/// Support for `__DATE__`, `__TIME__`, and `__TIMESTAMP__`
+struct TimeStampInfo
+{
+    // Note: Those properties need to be guarded by a call to `init`
+    // The API isn't safe, and quite brittle, but it was left this way
+    // over performance concerns.
+    // This is currently only called once, from the lexer.
+    __gshared char[11 + 1] date;
+    __gshared char[8 + 1] time;
+    __gshared char[24 + 1] timestamp;
```

Comment:
Can't we make these `static immutable` and initialize them exactly once using a shared module constructor?

Reply:
I don't think this is the case here, since `initialize` method is a call that can be used on DMD as a library and initialization can happen only when required.

Reply:
I think we can, however, since we are touching here, add small comments explaining the numbers